### DISCUSSION TITLE
The task was stuck, but when marking it as successful, it continues to run the remaining part of the job.

### DISCUSSION THREAD

**Discussion by Nguyenlthoe** [apache/airflow#40340] [General] [answered]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.9.0

### What happened?

I have a task with two main functions: performing a loop to catch exceptions and print them to the console, then sending a message to Telegram. But task stucked in the n-th iteration. Then I marked it as success in web UI, this task continued running in server and i received message from it in telegram.
Config: 
* Airflow HA with celery worker.
* Task using SshHook() connector to run job in another server.

### What you think should happen instead?

My DAG is using two SshHook Connector, I think airflow kill process with pid in celery worker but process in server ssh still running.

### How to reproduce

- create DAG with two tasks executed in order, where each task uses an SSH connection to a different server to perform its job.
- Task 1: performing a loop to catch exceptions and print them to the console, then sending a message to Telegram.
- Run DAG.

### Operating System

NAME="Ubuntu" VERSION="18.04.4 LTS (Bionic Beaver)" 

### Versions of Apache Airflow Providers

apache-airflow-providers-celery==3.6.0
apache-airflow-providers-common-io==1.3.1
apache-airflow-providers-common-sql==1.10.1
apache-airflow-providers-fab==1.0.4
apache-airflow-providers-ftp==3.7.0
apache-airflow-providers-http==4.9.0
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-papermill==2.0.1
apache-airflow-providers-smtp==1.6.1
apache-airflow-providers-sqlite==3.7.0
apache-airflow-providers-ssh==2.4.4
apache-airflow-providers-telegram==2.0.1
apache-airflow-providers-trino==3.1.0

### Deployment

Other

### Deployment details

Ubuntu

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
Likely you need to implement proper `on_kill` implementation that will perform cleanup. It's speciic to the operator you are running, if you are implementing in task flow , I do not think there is a way to create on_kill for such decorated task so that would be a new feature to add. Simple workaround to it is to write your own custom operator and use it instead of task flow decorated method (and implement on_kill there). I convert this one into discussion if needed. But those are basically two ways to address it:

* add PR proposing how to add on_kill to decorated task (new feature)
* or convert your task to classic operator and implement on_kill

**Comment by potiuk** [comment, marked as answer]
Also In case you use SSHOperator - I  added a feature issue and marked it as `good first issue` as I noticed it did not have one https://github.com/apache/airflow/issues/40343 - that might help in some cases. If you feel like it you might want to implement it (and if you are using SSHOperator it might also help to close connections in a gentle way).

**Reply by Nguyenlthoe** [reply]
Thank for your help.
